src/Compiler/test/Microsoft.AspNetCore.Razor.Test.Common/Language/ImmutableArrayExtensions.cs

```diff
@@ -0,0 +1,20 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Immutable;
+
+namespace Microsoft.CodeAnalysis.Razor;
+
+/// <summary>
+/// <see cref="ImmutableArray{T}"/> extension methods
+/// </summary>
+internal static class ImmutableArrayExtensions
```

Comment:
Let's put this in one of the main src projects, rather than the common test layer. We'll almost certainly want to end up using this extension in main product code too.

Reply:
There's a new `Microsoft.AspNetCore.Razor.Common` project that might be a good spot. It should referenced transitively by everything in razor-tooling.

Reply:
Not sure thats in the `the-merge` branch, but I fully support this idea. There are lots of little helpers I miss from Roslyn.

If we're going to snap early and merge, we could just move this after then.

Reply:
@333fred is already working on sharing `SyntaxNode` by source. If there are other helpers, more collection level utilities should we create a second source share NuPkg that includes those? 

I'd suggest a DLL but that means we'd need a VS perf exception.

Reply:
Now that `main` [has been merged](https://github.com/dotnet/razor-tooling/pull/6943) into `the-merge`, I've moved the `ImmutableArrayExtensions` into `Microsoft.AspNetCore.Razor.Common`: 45808eb6fc95a36080c3c42d0e3f6e8d9867c93d